server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -879,7 +884,11 @@ private void executePipelines(
         ingestDocument.resetReroute();
         final String originalIndex = indexRequest.indices()[0];
         final Consumer<Exception> exceptionHandler = (Exception e) -> {
+            if (originalIndex != null) {
+                FailureStoreMetrics.incrementForIndex(failureStoreMetrics.failureCounter(), originalIndex);
+            }
```

Comment:
Apparently, during a bulk upsert ([example](https://github.com/nielsbauman/elasticsearch/blob/46770e7af2c0146ac411376d566d5d0d1791b770/modules/ingest-common/src/yamlRestTest/resources/rest-api-spec/test/ingest/60_fail.yml#L105-L105)) we don't have an index name in the index request.